pkg/cluster/pod.go

```diff
@@ -399,11 +400,12 @@ func (c *Cluster) getPatroniMemberData(pod *v1.Pod) (patroni.MemberData, error)
 }
 
 func (c *Cluster) recreatePod(podName spec.NamespacedName) (*v1.Pod, error) {
+	stopCh := make(chan struct{})
 	ch := c.registerPodSubscriber(podName)
 	defer c.unregisterPodSubscriber(podName)
-	stopChan := make(chan struct{})
+	defer close(stopCh)
 
-	err := retryutil.Retry(c.OpConfig.PatroniAPICheckInterval, c.OpConfig.PatroniAPICheckTimeout,
+	err := retryutil.Retry(c.OpConfig.ResourceCheckInterval, c.OpConfig.PodDeletionWaitTimeout,
```

Comment:
again the timeout values is increased from 5s to 10 minutes

Reply:
PatroniAPICheckTimeout is the wrong timeout and to me PodDeletionWaitTimeout looked more meaning full. But since it's not a waiting but deleting maybe we should go back to the previous hard-coded value of 5 seconds.